### ISSUE TITLE
Does Batchnorm run correctly on reexecuted traces?

### ISSUE THREAD

**Issue by ezyang** [ezyang/pytorch-unattached#81]
I am not sure, since the C++ object has some stashed mutable state. Maybe it's OK because batchnorm is always used as a Module, and the Module is persistent across training iterations.

**Comment by zdevito** [comment]
I believe the stashed state is only used during testing. During training it always just computes the norm over the batch. So we are getting started by just running both in training model to compare.

**Comment by prigoyal** [comment]
close this?
